src/main/java/com/google/devtools/build/lib/rules/proto/ProtoInfo.java

```diff
@@ -0,0 +1,64 @@
+
+package com.google.devtools.build.lib.rules.proto;
+
+import com.google.devtools.build.lib.actions.Artifact;
+import com.google.devtools.build.lib.concurrent.ThreadSafety.Immutable;
+import com.google.devtools.build.lib.events.Location;
+import com.google.devtools.build.lib.skyframe.serialization.autocodec.AutoCodec;
+import com.google.devtools.build.lib.skyframe.serialization.autocodec.AutoCodec.VisibleForSerialization;
+import com.google.devtools.build.lib.skylarkbuildapi.proto.ProtoInfoApi;
+import com.google.devtools.build.lib.syntax.Environment;
+import com.google.devtools.build.lib.syntax.EvalException;
+import com.google.devtools.build.lib.packages.BuiltinProvider;
+import com.google.devtools.build.lib.packages.NativeInfo;
+
+@Immutable
+@AutoCodec
+public final class ProtoInfo extends NativeInfo implements ProtoInfoApi<Artifact> {
```

Comment:
Why do you need to create a new provider? There is already one that tells which source files a `proto_library` provides, `ProtoSourcesProvider`.